ee/session_recordings/queries/test/test_session_recording_list_from_session_replay.py

```diff
@@ -0,0 +1,422 @@
+        self.assertQueryMatchesSnapshot(generated_query)
+
+    @override_settings(
+        PERSON_ON_EVENTS_OVERRIDE=False, PERSON_ON_EVENTS_V2_OVERRIDE=True, ALLOW_DENORMALIZED_PROPS_IN_LISTING=True
+    )
+    def test_poe_v2_available_person_properties_are_used_in_replay_listing(self) -> None:
+        assert self.team.person_on_events_mode == PersonOnEventsMode.V2_ENABLED
+        materialize("events", "rgInternal", table_column="person_properties")
+
+        filter = SessionRecordingsFilter(
+            team=self.team,
+            data={
+                "properties": [
+                    {
+                        "key": "rgInternal",
+                        "value": ["false"],
+                        "operator": "exact",
+                        "type": "person",
+                    }
+                ]
+            },
+        )
+        session_recording_list_instance = SessionRecordingListFromReplaySummary(filter=filter, team=self.team)
+        [generated_query, query_params] = session_recording_list_instance.get_query()
+        assert query_params == {
+            "clamped_to_storage_ttl": mock.ANY,
+            "end_time": mock.ANY,
+            "event_end_time": mock.ANY,
+            "event_names": [],
+            "event_start_time": mock.ANY,
+            "kglobal_0": "rgInternal",
+            "limit": 51,
+            "offset": 0,
+            "start_time": mock.ANY,
+            "team_id": self.team.id,
+            "vglobal_0": ["false"],
+        }
+
+        # the unmaterialized column should query should not be used
+        assert (
+            "has(%(vperson_filter_pre__0)s, replaceRegexpAll(JSONExtractRaw(properties, %(kperson_filter_pre__0)s)"
+            not in generated_query
+        )
+        assert 'AND (  has(%(vglobal_0)s, "mat_pp_rgInternal"))' in generated_query
+        self.assertQueryMatchesSnapshot(generated_query)
+
+    @parameterized.expand(
+        [
+            ["poe and materialized columns allowed", True, True],
+            ["poe and materialized columns off", True, False],
+            ["poe off and materialized columns allowed", False, True],
+            ["neither poe nor materialized columns", False, False],
+        ]
+    )
+    @snapshot_clickhouse_queries
+    def test_event_filter_with_person_properties_materialized(
+        self, _name: str, poe2_enabled: bool, allow_denormalised_props: bool
+    ) -> None:
+        # KLUDGE: I couldn't figure out how to use @also_test_with_materialized_columns(person_properties=["email"])
+        # KLUDGE: and the parameterized.expand decorator at the same time, so I'm manually materializing here
+        materialize("events", "email", table_column="person_properties")
+        materialize("person", "email")
+
+        with self.settings(
+            PERSON_ON_EVENTS_V2_OVERRIDE=poe2_enabled, ALLOW_DENORMALIZED_PROPS_IN_LISTING=allow_denormalised_props
```

Comment:
Ah looks like we do it here! We should copy this approach to simplify the two tests above